.changeset/shy-dots-prove.md

```diff
@@ -0,0 +1,59 @@
+---
+"fluid-framework": minor
+"@fluidframework/tree": minor
+---
+---
+"section": tree
+---
+
+Improve strictness of input tree types when non-exact schema are provided
+
+When the type of the schema is not exactly specified, for example:
+
+```typescript
+const schemaFactory = new SchemaFactory("com.myApp");
+class A extends schemaFactory.object("A", {}) {}
+class B extends schemaFactory.array("B", schemaFactory.number) {}
+
+// Gives imprecise type (typeof A | typeof B)[]
```

Comment:
I couldn't really understand why the suggestions to remove the ambiguity worked until I got to the end of the changeset and saw a comment pointing out this is the "correct" type, so pointing it out here seems good to make it clear from the start.
```suggestion
// Gives imprecise type (typeof A | typeof B)[], where [typeof A, typeof B] is what would work best for Tree
```